packages/cms/src/lokalize/sync-after-feature.ts

```diff
@@ -0,0 +1,178 @@
+    /**
+     * No need to query documents if there are no mutations
+     */
+    return;
+  }
+
+  const allDevTexts = (await getClient('development')
+    .fetch(`*[_type == 'lokalizeText'] |
+  order(subject asc)`)) as LokalizeText[];
+
+  await applyDeletionsToDevelopment(deletions, allDevTexts);
+
+  await syncAdditionsToProduction(additions, allDevTexts);
+})().catch((err) => {
+  console.error('An error occurred:', err.message);
+  process.exit(1);
+});
+
+async function syncAdditionsToProduction(
+  additions: TextMutation[],
+  allTextDocuments: LokalizeText[]
+) {
+  if (additions.length === 0) {
+    console.log('There are no mutations that result in keys to add');
+    process.exit(0);
+  }
+
+  const prdTransaction = getClient('production').transaction();
+
+  let successCount = 0;
+  let failureCount = 0;
+
+  for (const addition of additions) {
+    const document = allTextDocuments.find((x) => x.key === addition.key);
+
+    if (document) {
+      const documentToInject: LokalizeText = {
+        ...document,
+        /**
+         * At this point we can not assume that this flag is still set like it
+         * was when the CLI command added the document, because in the meantime
+         * we could hit publish in development which clears the flag. So all
+         * text that are injected into production get this flag set here to be
+         * sure they show up as "new" there.
+         */
+        is_newly_added: true,
+        /**
+         * To know how often communication changes these texts we need to clear
+         * the counter from any publish actions we did ourselves in development.
+         */
+        publish_count: 0,
+      };
+
+      /**
+       * Using createIfNotExist we can safely run the script multiple times and
+       * we will never overwrite what has already been edited in the production
+       * set.
+       */
+      prdTransaction.createIfNotExists(documentToInject);
+      successCount++;
+    } else {
+      /**
+       * This should never happen, but it is also not severe enough to
+       * completely halt the script
+       */
+      console.warn(
+        `An addition for key ${addition.key} was requested, but the document can not be found in the development dataset`
+      );
+
+      failureCount++;
+    }
+  }
+
+  await prdTransaction.commit();
+
+  if (failureCount === 0) {
+    console.log(`Successfully injected all ${successCount} text keys`);
+  } else {
+    console.log(
+      `Injected ${successCount} text keys. Failed to add ${failureCount}`
+    );
+  }
+}
+
+async function applyDeletionsToDevelopment(
+  deletions: TextMutation[],
+  allTextDocuments: LokalizeText[]
+) {
+  if (deletions.length === 0) {
+    console.log('There are no mutations that result in keys to delete');
+    process.exit(0);
+  }
+
+  let successCount = 0;
+  let failureCount = 0;
+
+  const documentIdsToDelete = deletions
+    .map(({ key }) => allTextDocuments.find((x) => x.key === key))
```

Comment:
Here's a bug because the query also includes drafts. Drafts should also be removed, and the client-facing bug is that delete only deletes a draft if that document is found earlier.

I think we should delete both drafts as documents.
```suggestion
    .flatMap(({ key }) => allTextDocuments.filter(x) => x.key === key))
```

Reply:
I wasn't aware that drafts are separate documents.

Reply:
yep I think they are basically copies of the original document but the `id` is prepended with `drafts.`

Reply:
I think the easiest solution is to just not inject drafts from development into production. I think the mental model is a bit easier then. Otherwise the draft and published documents will both show up in the list of new documents on production.